### PR TITLE
feat(config): customizable configurations

### PR DESCRIPTION
**Pull request by jonzxz**
1. Added `Configuration` struct to be able to define configurations
2. Added `DefaultConfigurations` struct constructor function to return default configurations
3. Added `GetConfigurations` struct function to either
    1. create `ucd.conf` in `~/.config/ucd/` if does not exist and populate with `DefaultConfigurations`. 
    2. populate contents of existing `ucd.conf` into a `Configuration` struct to be used.

### CODE REVIEW

File: configurations/configurations.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		defer configFile.Close()
+
+		defaultConfigs, err := json.MarshalIndent(configurations, "", "\t")
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		configFile.Write(defaultConfigs)
+		configFile.Sync()
+		configFile.Close()
```
Comment (ztcjoe93) [inline configurations/configurations.go]: We can omit the `.Close()` call here -- the call is deferred at line 40 so the file will be closed once it's out of scope

File: configurations/configurations.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			log.Fatalln(err)
+		}
+		defer configFile.Close()
+
+		defaultConfigs, err := json.MarshalIndent(configurations, "", "\t")
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		configFile.Write(defaultConfigs)
```
Comment (ztcjoe93) [inline configurations/configurations.go]: We can use [os#WriteFile](https://pkg.go.dev/os#WriteFile) here to circumvent the need to explicitly create the configFile

File: configurations/configurations.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	"os"
+)
+
+type Configuration struct {
+	MaxMRUDisplay string `json:"MaxMRUDisplay"`
+}
+
+func DefaultConfigurations() Configuration {
+	return Configuration{
+		MaxMRUDisplay: "10",
```
Comment (ztcjoe93) [inline configurations/configurations.go]: Probably want to have `int` type for this?

**Review by ztcjoe93 — COMMENTED** [review]
While reviewing I was wondering if we can make the flow abit streamlined, perhaps something along the lines of:  

pseudo-code  
**os.MkdirAll to create if the `.config/ucd` directory doesn't exist**  
**readAll against the file**
**check byte count of file, if more than 0 we unmarshal it into the struct (indicates that something exists)**
**convert struct into byte[], write into file**

File: configurations/configurations.go (hunk 1 above)
Comment (jonzxz) [reply configurations/configurations.go]: Removed

File: configurations/configurations.go (hunk 2 above)
Comment (jonzxz) [reply configurations/configurations.go]: Replaced with WriteFile

File: configurations/configurations.go (hunk 3 above)
Comment (jonzxz) [reply configurations/configurations.go]: Fixed.

**Comment by jonzxz** [comment]
> While reviewing I was wondering if we can make the flow abit streamlined, perhaps something along the lines of:
> 
> pseudo-code **os.MkdirAll to create if the `.config/ucd` directory doesn't exist** **readAll against the file** **check byte count of file, if more than 0 we unmarshal it into the struct (indicates that something exists)** **convert struct into byte[], write into file**

Refactored to above logic to MkdirAll if os.Open returns an error.

Invokes `ReadAll` to retrieve bytes of files and does a check to read configs if `ReadAll` returns contents, otherwise marshal in default configs.

**Comment by ztcjoe93** [comment]
LGTM! 🚀

**Pull request merged**